client/components/Carousel.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+// hase a  x button on click changes modal click to false
+// has left right arrows
+// has main image
+// has caro
+class Carousel extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      images: this.props.images,
+      currentView: this.props.currentView,
```

Comment:
React docs says to avoid copying props inside state: https://reactjs.org/docs/react-component.html#constructor

One suggestion that I can make is that, if you won't change the this.state.images, you can use the images directly in your render method.

Something like that in the 24th line: `this.props.images.map(...)`

If I understood, every place will have some custom images, right? So they won't change.
As they won't change, you don't need to hold this data inside the state.

But this is just a suggestion.
If this is working in your view, skip this comment.

Reply:
Thanks this is  a really helpful tip I didn't know that. I jus updated  it. I my reasoning behind it was to  have a reference on the page. Like you have to look to another page to see the props.